frontend/src/components/layout/UserDropDownContainer.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import UserDropDown from './UserDropDown'
+
+interface Props {
+  user?: {
+    role?: string,
+    unit?: string,
+    department?: string,
+    group?: string
+  };
+  avatarUrl?: string;
+  userName?: string;
+}
+
+const UserDropDownContainer: React.FC<Props> = ({ user, avatarUrl, userName }) => {
+
+  const [open, setUserOpen] = React.useState(false);
+  const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);
+
+  const openUserInfo = (event: any) => {
+    const currentState = open;
+    setUserOpen(!currentState);
```

Comment:
It's a wrong way to change state. Try to get current state from arrow function inside `setDropdownState` call